### PR TITLE
Migration support for older mupdf-js library

### PR DESCRIPTION
**Pull request by andytango**
This PR includes documentation for migrating from, and API-compatible functions with https://github.com/andytango/mupdf-js.

FAO @jamie-lemon

### CODE REVIEW

**Comment by andytango** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jamie-lemon** [comment]
@ccxvii Would you mind reviewing the changes in the mupdf typescript , wasm related and c files here?

**Review by jamie-lemon — CHANGES_REQUESTED** [review]
I wonder if it is worth getting some of the tasks function case sensitivity finessed and consistent?

For example:
`loadPdf` -> `loadPDF`
`drawPageAsPng` -> `drawPageAsPNG`
`drawPageAsSvg` -> `drawPageAsSVG`
`drawPageAsHtml` -> `drawPageAsHTML`

**Comment by ccxvii** [comment]
drawPageAsSvg can be implemented using plain JS without needing to add a C function:

```
export function drawPageAsSvg(document: mupdf.PDFDocument, pageNumber: number): string {
        const page = document.loadPage(pageNumber)
        const buf = new mupdf.Buffer()
        const wri = new mupdf.DocumentWriter(buf, "svg", "")
        const dev = wri.beginPage(page.getBounds())
        page.run(dev, mupdf.Matrix.identity)
        dev.close()
        wri.endPage()
        return buf.asString()
}
```

**Comment by jamie-lemon** [comment]
@andytango If you can fix up the case sensitivity - see: https://github.com/ArtifexSoftware/mupdf.js/pull/105#pullrequestreview-2213952738 and do `drawPageAsSVG` as Tor suggests without the need for the C function - https://github.com/ArtifexSoftware/mupdf.js/pull/105#issuecomment-2273849892 then I think we are all good.

**Comment by andytango** [comment]
@jamie-lemon  PR updated as requested

File: docs/how-to-guide/migration/index.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+----
+
+2. Loading a document
+-------------------
+
+Just like with `mupdf-js`, you can load a document either as a Buffer
+(in Node.js), an ArrayBuffer (in the browser), or a Uint8Array (in both environments).
+
+We provide a `loadPdf` function that is similar to the `load` method in `mupdf-js`.
```
Comment (jamie-lemon) [inline docs/how-to-guide/migration/index.rst:37]: `loadPdf` -> `loadPDF`

File: docs/how-to-guide/migration/index.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      const arrayBuf = new Uint8Array(buf);
+      const doc = mupdf.load(arrayBuf);
+    }
+----
+
+With this:
+
+.. code-block:: javascript
+
+    import { loadPdf } from "mupdf/tasks";
```
Comment (jamie-lemon) [inline docs/how-to-guide/migration/index.rst:56]: `loadPdf` -> `loadPDF`

File: docs/how-to-guide/migration/index.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    async function handleSomePdf(file) {
+      const buf = await file.arrayBuffer();
+      const arrayBuf = new Uint8Array(buf);
+      const doc = loadPdf(arrayBuf); // Returns a Document instance
```
Comment (jamie-lemon) [inline docs/how-to-guide/migration/index.rst:61]: `loadPdf` -> `loadPDF`

**Review by jamie-lemon — CHANGES_REQUESTED** [review]

**Review by jamie-lemon — APPROVED** [review]

**Pull request merged**